Upgrade packages in toolbelt-boshrelease to newest versions

noahccoker opened the pull request:
Complete updated list of package versions:

- toolbelt-boss
	- boss version:	v0.0.6
- toolbelt-cf
	- cf version:	7.4.0+e55633fed.2021-11-15
	- cf6 version:	6.53.0+8e2b70a4a.2020-10-01
	- cf7 version:	7.4.0+e55633fed.2021-11-15
	- cf8 version:	8.1.0+034d929d7.2022-01-03
- toolbelt-cfdot
	- N/A
- toolbelt-esuf
	- esuf version:	v0.1.2
- toolbelt-gaol
	- N/A
- toolbelt-hatop
	- hatop version:	0.8.0
	- Switched to more recently updated repo:
		- link: https://github.com/jhunt/hatop/releases/tag/v0.8.1
	- Need to raise issue
		- Issue: hatop script relies on /usr/bin/python, but release
			-		currently has /usr/bin/python3 by default without
			-		alias to /usr/bin/python
- toolbelt-jq
	- jq version: jq-1.6
- toolbelt-mysql-client
	- mysql version:	Ver 15.1 Distrib 10.6.5-MariaDB
- toolbelt-nats
	- nats version: 1.2.0
	- Need to raise issue
		- Issue: App is hard-coded to print version 1.0.2
			- even though the app is on version 1.2.0
				- See issue on Github: https://github.com/starkandwayne/nats/issues/10
- toolbelt-netsniff-ng
	- netsniff-ng version:	0.6.8
		- ncurses version: ncurses-6.3
				- link: https://ftp.gnu.org/pub/gnu/ncurses/
		- libsodium
			- version: libsodium-1.0.18.tar.gz
			- newer version: N/A
				- link: https://download.libsodium.org/libsodium/releases/libsodium-1.0.18.tar.gz
		- libnfnetlink
			- version: libnfnetlink-1.0.1.tar.bz2
			- newer version: N/A
				- link: https://www.netfilter.org/projects/libnfnetlink/index.html
		- libnl
			- version: libnl-3.2.25.tar.gz
			- newer version: N/A
				- link: https://www.infradead.org/~tgr/libnl/files/
		- libmnl
			- version: libmnl-1.0.4.tar.bz2
			- newer version: N/A
				- link: https://www.netfilter.org/projects/libmnl/files/
		- libnetfilter-conntrack
			- version: libnetfilter\_conntrack-1.0.8.tar.bz2
			- newer version: N/A
				- link: https://www.netfilter.org/projects/libnetfilter_conntrack/files/
		- libpcap
			- version: libpcap-1.9.1.tar.gz
			- newer version: 1.10.1.tar.gz
				- link: https://www.tcpdump.org/release/
		- userspace-rcu
			- version: userspace-rcu-0.12.0.tar.bz2
			- newer version: userspace-rcu-0.13.1.tar.bz2
				- link: https://lttng.org/files/urcu/
		- libcli
			- version: libcli-1.10.4.tar.gz
			- newer version: V1.10.7
				- link: https://github.com/dparrish/libcli/
		- libnet
			- version: libnet-1.2-rc3.tar.gz
			- newer version: N/A
				- link: https://sourceforge.net/projects/libnet-dev/
- toolbelt-nload
	- nload version:	N/A
- toolbelt-nmap
	- nmap version:	7.92
- toolbelt-psql
	- psql version:	14.1
- toolbelt-redis
	- redis version:	6.2.6
- toolbelt-safe
	- safe version:	v1.6.1
- toolbelt-screen
	- screen version:	4.08.00
- toolbelt-spruce
	- spruce version:	1.29.0
- toolbelt-tcptrace
	- tcpdump version:	4.9.3
	- libpcap version:	1.8.1
	- OpenSSL version:	1.1.1
- toolbelt-tree
	- tree version:	
- toolbelt-tshark
	- tshark version:	(omitted from this release)
		- libffi
			- version: libffi-3.3.tar.gz 
			- newer version: N/A
				- link: https://sourceware.org/pub/libffi/
		- glib
			- version: glib-2.45.8.tar.xz
			- newer version: glib-2.71.0.tar.xz
				- link: https://download-fallback.gnome.org/sources/glib/
		- libpcap
			- version: libpcap-1.9.1.tar.gz
			- newer version: 1.10.1.tar.gz
				- link: https://www.tcpdump.org/release/
		- wireshark
			- version: wireshark-2.6.16.tar.xz
			- newer version: wireshark-3.6.1.tar.xz
				- link: https://2.na.dl.wireshark.org/src/
- toolbelt-vault
	- vault version:	v1.9.2
- toolbelt-veritas
	- veritas version:	N/A

ncoker-starkandwayne committed:
Updating package versions to latest

ncoker-starkandwayne committed:
Upgraded packages to latest versions

ncoker-starkandwayne committed:
Fixed toolbelt-cf; t-shark and python-release still WIP

ncoker-starkandwayne committed:
Commented out parts related to python-release; fixed issues with hatop, nats, and tree packaging scripts; changed blobstore back to s3; added notes about versions and download urls in comments of packaging scripts

norman-abramovitz commented on config/blobs.yml:
```diff
@@ -2,18 +2,26 @@ boss/boss-linux-amd64:
   size: 24614808
   object_id: b673e243-7c22-4fd3-5ec2-945d8557c4f1
   sha: sha256:3c4bfbab50aba7e21aa4d0617990f6f59c1ad391b092edf65f1cd5eae95741b7
 esuf/esuf-linux-amd64:
```
I am not seeing an objectid for esuf

norman-abramovitz commented on config/blobs.yml:
```diff
@@ -34,14 +48,26 @@ mysql/mariadb-10.4.12.tar.gz:
   size: 79120835
   object_id: 33c07e17-733f-4718-7d9b-c8f97583593a
   sha: sha256:fef1e1d38aa253dd8a51006bd15aad184912fce31c446bb69434fcde735aa208
+mysql/mariadb-10.6.5.tar.gz:
```
upload-blob is needed

norman-abramovitz commented on packages/toolbelt-cf/packaging:
```diff
@@ -1,12 +1,47 @@
+# cp cf7 ${BOSH_INSTALL_TARGET}/bin
+# chmod 755 ${BOSH_INSTALL_TARGET}/bin/cf7
+
+#(cd tmp/
```
Why did you comment out using a temporary subdirectory.   I would remove those lines.

norman-abramovitz commented on packages/toolbelt-hatop/packaging:
```diff
@@ -1,9 +1,21 @@
-set -eu
+set -xeu
+
+# link: https://github.com/jhunt/hatop/releases/tag/v0.8.1
```
I question this location for hatop unless we think James will continue to maintain this application.

norman-abramovitz commented on packages/toolbelt-hatop/packaging:
```diff
@@ -1,9 +1,21 @@
+
+# create a wrapper script to replace this that will add python 
+#   to the path when it runs
+if [[ ! -f /usr/bin/python ]] ; then 
```
What happens if there is no python3 at all?

norman-abramovitz commented on packages/toolbelt-cf/packaging:
```diff
@@ -1,12 +1,47 @@
+
+#(cd tmp/
+(
+tar -xzvf cf-cli/cf7-cli_7.4.0_linux_x86-64.tgz cf7 &&
```
This looks these lines are duplicates of lines 25 to 27.

ncoker-starkandwayne committed:
updated blobstore

cweibel commented:
Did a quick skim, only think you might want to change is removing the commented rows out of the sample manifest to clean it up just a little.

norman-abramovitz commented on config/blobs.yml:
```diff
@@ -26,6 +38,14 @@ hatop/hatop:
+  size: 3953824
+  object_id: d5ccf0ea-0ccc-4cf7-648e-593fe39e0c16
+  sha: sha256:af986793a515d500ab2d35f8d2aecd656e764504b789b66d7e1a0b727a124c44
 jq/jq-linux64:
```
Old packages should be removed since we do not want them to be downloaded when the new release is created.

Easiest way is to delete the entries in blobs.yml.    What I would like to do is to create a file with the old entries so we can map back to what is in the S3 bucket.

norman-abramovitz commented on manifests/toolbelt.yml:
```diff
@@ -12,16 +12,16 @@ instance_groups:
-      - { name: toolbelt,            release: toolbelt }
-      - { name: toolbelt-quick,      release: toolbelt }
+#  - name: quick
+#    instances: 1
```
This needs to be undone

norman-abramovitz commented on manifests/toolbelt.yml:
```diff
@@ -37,6 +37,9 @@ stemcells:
-  version: 3.5.1
-  url:     https://github.com/cloudfoundry-community/toolbelt-boshrelease/releases/download/v3.5.1/toolbelt-3.5.1.tgz
-  sha1:    347c735315678cbb1e21c2412601dac24af82501
+  version: latest
```
This entry should be updated with the new release and version.

ncoker-starkandwayne committed:
Fixed small issues found during the review of the pull request;
Added to the redis packaging script to include v5 and v6;
List of outdated packages that were removed from the blobs.yml file:

cf-cli/cf-linux64:
  size: 24682496
  object_id: 83d125bf-149f-40ed-7475-e573f2cd5aff
  sha: sha256:099a5a45c8817ad56222b9e5253282ba972ae4dfe639708d4e1d8360552ce162
hatop/hatop:
  size: 72429
  object_id: 9d096d6c-5f55-456a-8387-39d45f3d7a0e
  sha: daddb3f13a70d4e2fa0802333c2e13468635171f
jq/jq-linux64:
  size: 3953824
  object_id: a3306979-07fc-4198-5722-34e0bc7f416e
  sha: sha256:af986793a515d500ab2d35f8d2aecd656e764504b789b66d7e1a0b727a124c44
mysql/mariadb-10.4.12.tar.gz:
  size: 79120835
  object_id: 33c07e17-733f-4718-7d9b-c8f97583593a
  sha: sha256:fef1e1d38aa253dd8a51006bd15aad184912fce31c446bb69434fcde735aa208
nats/nats-1.0.2-linux-x86_64.tar.gz:
  size: 1976167
  object_id: 9530f4aa-fcca-43d2-af55-34b8b44a728c
  sha: a47897ac81b83d7e72344ef0c58b90dc9db036c5
netsniff-ng/libcli-1.10.4.tar.gz:
  size: 53125
  object_id: 0f36d8e9-7f72-417a-543b-1a77479814c2
  sha: sha256:43efade6f3deb621e1e226b7854274a869fca57e12e74e61c5496d5dea7c445d
netsniff-ng/libpcap-1.9.1.tar.gz:
  size: 861228
  object_id: d79c48fb-3c23-41a2-7003-31d4e6cd347d
  sha: sha256:635237637c5b619bcceba91900666b64d56ecb7be63f298f601ec786ce087094
netsniff-ng/ncurses-6.2.tar.gz:
  size: 3425862
  object_id: 88a250a0-6a3c-4d39-7dd1-d54985a74715
  sha: sha256:30306e0c76e0f9f1f0de987cf1c82a5c21e1ce6568b9227f7da5b71cbea86c9d
netsniff-ng/netsniff-ng-0.6.6.tar.gz:
  size: 802394
  object_id: f38718ae-7135-4c06-5733-eb2b62f1f227
  sha: 956f3b3a5f871330819f96abfeede7923d8c557c
netsniff-ng/userspace-rcu-0.12.0.tar.bz2:
  size: 528838
  object_id: 45f7bc4c-4190-442f-518f-a058a60820a8
  sha: sha256:409b1be506989e1d26543194df1a79212be990fe5d4fd84f34f019efed989f97
nmap/nmap-7.80.tar.bz2:
  size: 10550327
  object_id: 582cc4dc-630e-4933-4099-63bc777212c9
  sha: sha256:fcfa5a0e42099e12e4bf7a68ebe6fde05553383a682e816a7ec9256ab4773faa
postgresql/postgresql-11.7.tar.gz:
  size: 26065946
  object_id: 374a9aeb-b975-4954-5310-5cc59980a6cf
  sha: sha256:4db579e34f1d61433db9f29971b04934ea1e3318e4c3905b1e6f240e5b853539
tree/tree-1.8.0.tgz:
  size: 50286
  object_id: 7f12b2a4-3e0d-49b4-5256-5d591a42e58a
  sha: sha256:715d5d4b434321ce74706d0dd067505bb60c5ea83b5f0b3655dae40aa6f9b7c2
tshark/glib-2.45.8.tar.xz:
  size: 7188216
  object_id: 833a983c-402a-47de-b2ec-34622b701239
  sha: a25590b540251f9ed4a56ea2837d28071bddfa5a
tshark/libffi-3.3.tar.gz:
  size: 1305466
  object_id: fcf9d039-0863-41b9-7bf1-9dd61db42252
  sha: sha256:72fba7922703ddfa7a028d513ac15a85c8d54c8d67f55fa5a4802885dc652056
tshark/libpcap-1.9.1.tar.gz:
  size: 861228
  object_id: b9475b34-7fdd-441c-72e9-746b5ab050df
  sha: sha256:635237637c5b619bcceba91900666b64d56ecb7be63f298f601ec786ce087094
vault/vault_1.4.0_linux_amd64.zip:
  size: 48230599
  object_id: 0d91ce11-2220-47b4-50e6-0a397376f4ca
  sha: sha256:8f739c4850bab35e971e27c8120908f48f247b07717d19aabad1110e9966cded

ncoker-starkandwayne committed:
Fixed formatting in manifest file; Updated toolbelt-redis spec file to include v5 and v6

ncoker-starkandwayne committed:
fixed a bug in the redis packaging script

ncoker-starkandwayne committed:
small tweak to symlink in redis packaging script

ncoker-starkandwayne committed:
removed -v on tar command in cf packaging script

ncoker-starkandwayne committed:
added line to EOF for hatop packaging scipt

ncoker-starkandwayne committed:
added line to EOF for nats packaging scipt

ncoker-starkandwayne committed:
added line to EOF for redis packaging scipt

ncoker-starkandwayne committed:
added line to EOF for tshark packaging scipt

norman-abramovitz reviewed (approved):
Did a few reviews with Noah and Ben.  Ben did the final review and now the code looks ok. 
We are leaving the tshark for a later update since that is not needed for travelers.
